CFile.js

```diff
@@ -0,0 +1,229 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var fs = require("fs");
+var path = require("path");
+var log4js = require("log4js");
+
+var logger = log4js.getLogger("loctool.plugin.CFile");
+
+/**
+ * Create a new java file with the given path name and within
+ * the given project.
+ *
+ * @param {Project} project the project object
+ * @param {String} pathName path to the file relative to the root
+ * of the project file
+ * @param {FileType} type the file type of this instance
+ */
+var CFile = function(props) {
+    this.project = props.project;
+    this.pathName = props.pathName;
+    this.type = props.type;
+    this.API = props.project.getAPI();
+
+    this.set = this.API.newTranslationSet(this.project ? this.project.sourceLocale : "zxx-XX");
+};
+
+/**
+ * Unescape the string to make the same string that would be
+ * in memory in the target programming language.
+ *
+ * @static
+ * @param {String} string the string to unescape
+ * @returns {String} the unescaped string
+ */
+CFile.unescapeString = function(string) {
+    var unescaped = string;
+
+    unescaped = unescaped.
+        replace(/\\\\n/g, "").                // line continuation
+        replace(/\\\n/g, "").                // line continuation
+        replace(/^\\\\/, "\\").             // unescape backslashes
+        replace(/([^\\])\\\\/g, "$1\\").
+        replace(/^\\'/, "'").               // unescape quotes
+        replace(/([^\\])\\'/g, "$1'").
+        replace(/^\\"/, '"').
+        replace(/([^\\])\\"/g, '$1"');
+
+    return unescaped;
+};
+
+/**
+ * Clean the string to make a resource name string. This means
+ * removing leading and trailing white space, compressing
+ * whitespaces, and unescaping characters. This changes
+ * the string from what it looks like in the source
+ * code but increases matching.
+ *
+ * @static
+ * @param {String} string the string to clean
+ * @returns {String} the cleaned string
+ */
+CFile.cleanString = function(string) {
+    var unescaped = CFile.unescapeString(string);
+
+    unescaped = unescaped.
+        replace(/\\[btnfr]/g, " ").
+        replace(/[ \n\t\r\f]+/g, " ").
+        trim();
+
+    return unescaped;
+};
+
+/**
+ * Make a new key for the given string. This must correspond
+ * exactly with the code in htglob jar file so that the
+ * resources match up. See the class IResourceBundle in
+ * this project under the java directory for the corresponding
+ * code.
+ *
+ * @private
+ * @param {String} source the source string to make a resource
+ * key for
+ * @returns {String} a unique key for this string
+ */
+CFile.prototype.makeKey = function(source) {
+    return CFile.unescapeString(source);
+};
+
+var reGetLocString = new RegExp(/\bresBundle_getLocString\((.*)\,\s*[\"|\'](.*)*[\"|\']\)\;/g);
```

Comment:
This will not handle the case where there is an escaped double-quote character in the middle of the string. Instead, use a regular expression like:

```
"((\\"|[^"])*)"
```

to catch those.

Reply:
Also, C does not support single-quoted strings.

Reply:
I remove all the test cases with single-quote strings.
and I don't get that case. (escaped double-quote character in the middle of the string) Could you give me an example? Do you mean the following case?  https://github.com/iLib-js/ilib-loctool-webos-c/blob/initialSetup/test/testCFile.js#L237

Reply:
Yes, like that.

Reply:
look like current Regular expression is covering it. don't you think so?

Reply:
The case where there are two strings to extract on the same line will make this regular expression fail.

```
resBundle_getLocString(foo, "bar"); resBundle_getLocString(asdf, "xxx");
```

Will return a source string `bar"); resBundle_getLocString(asdf, "xxx`

Reply:
I added a testcase to verify this feature.https://github.com/iLib-js/ilib-loctool-webos-c/blob/initialSetup/test/testCFile.js#L620